libdnf/repo/solv_repo.cpp

```diff
@@ -495,7 +500,7 @@ std::string SolvRepo::solv_file_name(const char * type) {
 
 
 std::filesystem::path SolvRepo::solv_file_path(const char * type) {
-    return std::filesystem::path(config.get_cachedir()) / solv_file_name(type);
+    return std::filesystem::path(config.get_cachedir()) / "solv" / solv_file_name(type);
```

Comment:
What about to create `solv` in `repodata` directory to ensure that solve data are replaced when `repodata` dir is replaced when new cache is generated. It will improve the performance because solv data will be not necessary to check checksum because the file will be not present. Alternatively it is possible to remove `solv` dir when metadata are regenerated. But may be it is not necessary at all because when cache is regenerated there must be locking mechanism to prevent that someone else is trying to use cache.